x-pack/plugins/lens/public/indexpattern_datasource/visualize_geo_field_button.tsx

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { Component, MouseEvent } from 'react';
+import { EuiButton } from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { getUiActions } from '../kibana_services';
+import {
+  visualizeGeoFieldTrigger,
+  VISUALIZE_GEO_FIELD_TRIGGER,
+} from '../../../../../src/plugins/ui_actions/public';
+
+interface Props {
+  indexPatternId: string;
+  fieldName: string;
+}
+
+interface State {
+  hasLoadedHref: boolean;
+  href: string | null;
+}
+
+export class VisualizeGeoFieldButton extends Component<Props, State> {
```

Comment:
This is unusual because we only have one React class component in Lens- everything else is a function component. I wouldn't block the PR on this one.